impl/src/main/java/org/jboss/weld/xml/BeansXmlParser.java

```diff
@@ -44,17 +44,28 @@
 public class BeansXmlParser {
 
     private final BeansXmlValidator beansXmlValidator;
+    // having this information here allows WFLY to create appropriate parser
+    private final BeanDiscoveryMode emptyBeansXmlDiscoveryMode;
 
     public BeansXmlParser() {
+        this(false);
+    }
+
+    public BeansXmlParser(boolean legacyEmptyBeansXmlTreatment) {
```

Comment:
I left this as boolean since it is coming from WFLY but it could be changed as well.

Reply:
`legacyEmptyBeansXmlTreatment` -> `emptyBeansXmlDiscoveryModeAll`?